### PR TITLE
Show paid subscription needed dialog when unapproved user tries to send mail

### LINKED ISSUE #2781: Silent failure on sending email when account is in PaidSubscriptionNeeded
How to reproduce:
1. Set account to approval status PaidSubscriptionNeeded
2. Login into that account
3. Try to send an email to another internal user
-> The send progress dialog is shown shortly but the email draft stays open and the email is not sent

AccessBlockedError occurs: could not send mail (blocked access)

Should not be possible to open mail editor in this state, should see the dialog instead

moved from server issue 630

### PR DESCRIPTION
**Pull request by johnbotris**
fix #2781

We don't totally block the user from opening up a mail editor dialog, because they should still be able to send an email to sales@tutao.de. Instead we just show the subscription needed dialog when they try to open, and then again when they try to send

### CODE REVIEW

File: src/mail/editor/MailEditor.js [hunk 1]
```diff
@@ -517,14 +517,13 @@ function createMailEditorDialog(model: SendMailModel, blockExternalContent: bool
...
 export function newMailEditor(mailboxDetails: MailboxDetail): Promise<Dialog> {
-	return checkApprovalStatus(false).then(sendAllowed => {
-		if (sendAllowed) {
-			return import("../signature/Signature")
-				.then(({appendEmailSignature}) => appendEmailSignature("", logins.getUserController().props))
-				.then((signature) => newMailEditorFromTemplate(mailboxDetails, {}, "", signature))
-		} else {
-			return Promise.reject(new PermissionError("not allowed to send mail"))
-		}
+	// We check approval status so as to get a dialog informing the user that they cannot send mails
```
[the remark this answers is not in the document]
Comment (johnbotris) [reply src/mail/editor/MailEditor.js:519]: checkApprovalStatus get's called again if they receive AccessBlockedError upon sending a mail:
https://github.com/tutao/tutanota/blob/523ad0150c7f4054e2f0885280c1250ec818936b/src/mail/editor/SendMailModel.js#L745-L751

the popup happens inside there

[the remark this answers is not in the document]
Comment (johnbotris) [reply src/mail/editor/MailEditor.js:519]: this makes me wonder if checkApprovalStatus should maybe not be in "LoginUtils" for pedantic semantic reasons

**Comment by johnbotris** [comment]
@charlag can you confirm that it's ok to drop explicit handling of connection error here? currently we only deal with it on IE, otherwise we just throw again, so it should be fine to remove the IE check now that we don't support it right?

File: src/login/LoginViewController.js [hunk 2]
```diff
@@ -154,56 +154,30 @@ export class LoginViewController implements ILoginViewController {
...
-			            return errorAction()
-		            })
-		            .catch(ConnectionError, e => {
-			            if (client.isIE()) {
-				            // IE says it's error code 0 fore some reason
-				            this.view.helpText = lang.get('loginFailed_msg')
-				            m.redraw()
+
+			            this.view.invalidCredentials =
+				            this.view.invalidCredentials || e instanceof BadRequestError || e instanceof NotAuthenticatedError
```
Comment (charlag) [inline src/login/LoginViewController.js:162]: Why is it OR with previous value? Can it go `false` then?

Comment (charlag) [reply src/login/LoginViewController.js:162]: I see, maybe just "if" and then not set it otherwise? it will be more like before.

Comment (johnbotris) [reply src/login/LoginViewController.js:162]: it's the same, i just thought it was nice this way. if we can prefer i can change to ifs

**Pull request merged**